src/GraphControl/Control/Grapher.cpp

```diff
@@ -389,4 +434,98 @@ namespace GraphControl
             e->Handled = true;
         }
     }
+
+    void Grapher::OnPointerWheelChanged(PointerRoutedEventArgs^ args)
+    {
+        PointerPoint^ currentPointer = args->GetCurrentPoint(/*relative to*/ this);
+
+        double delta = currentPointer->Properties->MouseWheelDelta;
+
+        // The maximum delta is 120 according to:
+        // https://docs.microsoft.com/en-us/uwp/api/windows.ui.input.pointerpointproperties.mousewheeldelta#Windows_UI_Input_PointerPointProperties_MouseWheelDelta
+        // Apply a dampening effect so that small mouse movements have a smoother zoom.
+        constexpr double scrollDamper = 0.15;
+        double scale = 1.0 + (abs(delta) / WHEEL_DELTA) * scrollDamper;
+
+        // positive delta if wheel scrolled away from the user
+        if (delta >= 0)
+        {
+            scale = 1.0 / scale;
+        }
+
+        // For scaling, the graphing engine interprets x,y position between the range [-1, 1].
+        // Translate the pointer position to the [-1, 1] bounds.
+        const auto& pos = currentPointer->Position;
+        const auto[centerX, centerY] = PointerPositionToGraphPosition(pos.X, pos.Y, ActualWidth, ActualHeight);
+
+        ScaleRange(centerX, centerY, scale);
+
+        args->Handled = true;
+    }
+
+    void Grapher::OnPointerPressed(PointerRoutedEventArgs^ e)
+	{
+        // Set the pointer capture to the element being interacted with so that only it
+        // will fire pointer-related events
+        CapturePointer(e->Pointer);
+	}
+
+    void Grapher::OnPointerReleased(PointerRoutedEventArgs^ e)
+	{
+        // Release the pointer
+        ReleasePointerCapture(e->Pointer);
+	}
+
+    void Grapher::OnPointerCanceled(PointerRoutedEventArgs^ e)
+    {
+        ReleasePointerCapture(e->Pointer);
+    }
+
+    void Grapher::OnManipulationDelta(ManipulationDeltaRoutedEventArgs^ args)
+    {
+        if (m_graph)
+        {
+            if (auto renderer = m_graph->GetRenderer())
+            {
+                const double width = ActualWidth;
+                const double height = ActualHeight;
+
+                const auto& translation = args->Delta.Translation;
+                double translationX = translation.X;
+                double translationY = translation.Y;
+                if (translationX != 0 || translation.Y != 0)
+                {
+                    // The graphing engine pans the graph according to a ratio for x and y.
+                    // A value of +1 means move a half screen in the positive direction for the given axis.
+                    // Convert the manipulation's translation values to ratios for the engine.
+                    translationX /= -width;
+                    translationY /= height;
+
+                    if (FAILED(renderer->MoveRangeByRatio(translationX, translationY)))
```

Comment:
Same comment here, why not use the function that handles this from earlier?

Reply:
Answered below